Index: src/main/java/agh/ics/oop/World.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package agh.ics.oop;\r\n\r\nimport agh.ics.oop.model.MoveDirection;\r\n\r\npublic class World {\r\n    public static void main(String[] args) {\r\n        System.out.println(\"Start\");\r\n        MoveDirection[] directions = OptionParser.directionparser(args);\r\n        run(directions);\r\n        System.out.println(\"Stop\");\r\n    }\r\n\r\n    public static void run(MoveDirection[] direction) {\r\n        for(MoveDirection d : direction) {\r\n            switch (d){\r\n                case f -> System.out.println(\"Zwierzak idzie do przodu\");\r\n                case b -> System.out.println(\"Zwierzak idzie do tyłu\");\r\n                case r -> System.out.println(\"Zwierzak skręca w prawo\");\r\n                case l -> System.out.println(\"Zwierzak skręca w lewo\");\r\n            }\r\n        }\r\n    }\r\n}\r\n
===================================================================
diff --git a/src/main/java/agh/ics/oop/World.java b/src/main/java/agh/ics/oop/World.java
--- a/src/main/java/agh/ics/oop/World.java	(revision e7b83c0dce03a6c13377712e0baa825d7aa314d7)
+++ b/src/main/java/agh/ics/oop/World.java	(date 1729511948753)
@@ -5,14 +5,14 @@
 public class World {
     public static void main(String[] args) {
         System.out.println("Start");
-        MoveDirection[] directions = OptionParser.directionparser(args);
+        MoveDirection[] directions = OptionParser.parsedirection(args);
         run(directions);
         System.out.println("Stop");
     }
 
-    public static void run(MoveDirection[] direction) {
-        for(MoveDirection d : direction) {
-            switch (d){
+    private static void run(MoveDirection[] directions) {
+        for(MoveDirection  direction : directions) {
+            switch (direction){
                 case f -> System.out.println("Zwierzak idzie do przodu");
                 case b -> System.out.println("Zwierzak idzie do tyłu");
                 case r -> System.out.println("Zwierzak skręca w prawo");
Index: src/main/java/agh/ics/oop/OptionParser.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package agh.ics.oop;\r\n\r\nimport agh.ics.oop.model.MoveDirection;\r\n\r\nimport java.util.ArrayList;\r\nimport java.util.List;\r\n\r\npublic class OptionParser {\r\n    public static MoveDirection[] directionparser(String[] args) {\r\n        List<MoveDirection> directions = new ArrayList<>();\r\n        for (String arg : args) {\r\n            try{\r\n            MoveDirection direction = MoveDirection.valueOf(arg);\r\n            directions.add(direction);\r\n            } catch (IllegalArgumentException e) {\r\n                System.out.println(arg + \" is not a valid move direction\");\r\n                throw new RuntimeException(e);\r\n            }\r\n        }\r\n        return directions.toArray(new MoveDirection[0]);\r\n    }\r\n}\r\n
===================================================================
diff --git a/src/main/java/agh/ics/oop/OptionParser.java b/src/main/java/agh/ics/oop/OptionParser.java
--- a/src/main/java/agh/ics/oop/OptionParser.java	(revision e7b83c0dce03a6c13377712e0baa825d7aa314d7)
+++ b/src/main/java/agh/ics/oop/OptionParser.java	(date 1729511826499)
@@ -6,7 +6,7 @@
 import java.util.List;
 
 public class OptionParser {
-    public static MoveDirection[] directionparser(String[] args) {
+    public static MoveDirection[] parsedirection(String[] args) {
         List<MoveDirection> directions = new ArrayList<>();
         for (String arg : args) {
             try{
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"ExternalStorageConfigurationManager\" enabled=\"true\" />\r\n  <component name=\"FrameworkDetectionExcludesConfiguration\">\r\n    <file type=\"web\" url=\"file://$PROJECT_DIR$\" />\r\n  </component>\r\n  <component name=\"ProjectRootManager\" version=\"2\" languageLevel=\"JDK_21\" default=\"true\" project-jdk-name=\"21\" project-jdk-type=\"JavaSDK\">\r\n    <output url=\"file://$PROJECT_DIR$/out\" />\r\n  </component>\r\n</project>
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
--- a/.idea/misc.xml	(revision e7b83c0dce03a6c13377712e0baa825d7aa314d7)
+++ b/.idea/misc.xml	(date 1729509988918)
@@ -4,7 +4,7 @@
   <component name="FrameworkDetectionExcludesConfiguration">
     <file type="web" url="file://$PROJECT_DIR$" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" default="true" project-jdk-name="21" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" project-jdk-name="21" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
 </project>
\ No newline at end of file
